services/director-v2/src/simcore_service_director_v2/modules/catalog.py

```diff
@@ -120,6 +121,17 @@ async def get_service_labels(
             return SimcoreServiceLabels.model_validate(resp.json())
         raise HTTPException(status_code=resp.status_code, detail=resp.content)
 
+    async def get_service_extras(
+        self, service_key: ServiceKey, service_version: ServiceVersion
+    ) -> ServiceExtras:
+        resp = await self.request(
+            "GET",
+            f"/services/{urllib.parse.quote_plus(service_key)}/{service_version}/extras",
+        )
+        if resp.status_code == status.HTTP_200_OK:
+            return ServiceExtras.model_validate(resp.json())
+        raise HTTPException(status_code=resp.status_code, detail=resp.content)
```

Comment:
why you are raising `HTTPExceptions` here?  Note that this is a service layer and now our controllers are not only `rest` but also `rpc`. Note that `HTTPException` will be only handled by the rest layer (i.e. `fastapi` knows how to convert that into an http response)  but is not intended for the `rpc`

I highly advice to review the error policy and handling of of the director-v2 ASAP!

Reply:
not sure that is used by RPC at the moment, but yeah for sure these should be updated. anyway I would not do this in this PR.

Reply:
This will be removed in the future. Will not put time into dealing with it now. Since this code will be moved to the dynamic-scheduler service.